### ISSUE TITLE
Consider making webpage sections collapsible. 

### ISSUE THREAD

**Issue by edonehoo** [patternfly/patternfly-org#3151]
Consider making sections on component pages collapsible. 

- This is an enhancement suggestion based on the [design system competitive analysis.](https://docs.google.com/presentation/d/161XsZ6NrUSgnCkaA1e_96oxWY72VsJAa2SVmtIEhWJg/edit#slide=id.g547716335e_0_260)
- For example, consider making prop subsections (such as https://www.patternfly.org/v4/components/table#props) collapsible to make it easier to navigate through long sections.
- This was an idea coming at the site as a first-time user, but may or may not make as much functional sense for more experienced site users.

**Comment by evwilkin** [comment]
@mcarrano completely feasible, very low level of effort at first glance.  The main challenges look to be confirming how we want it styled (should the expandable text replace the existing section headers?) and the interaction with the JumpLinks (all jump links show by default, but seem to highlight strangely & not work as intended when the corresponding sections are collapsed.)

I threw together a quick POC in a few minutes (https://github.com/patternfly/patternfly-org/pull/3152), which highlights some of these considerations:
- https://patternfly-org-pr-3152-v4.surge.sh/v4/components/about-modal

**Comment by mcarrano** [comment]
Yeah, those are great questions @evwilkin . I am going to transfer this back to design so we can take a deeper look at this.
